examples/tv-casting-app/android/App/app/src/main/jni/cpp/support/Converters-JNI.cpp

```diff
@@ -366,6 +366,162 @@ jobject convertLongFromCppToJava(uint64_t responseData)
     return env->NewObject(responseTypeClass, constructor, responseData);
 }
 
+chip::Protocols::UserDirectedCommissioning::TargetAppInfo * convertTargetAppInfoFromJavaToCpp(jobject jTargetAppInfo)
+{
+    ChipLogProgress(AppServer, "convertTargetAppInfoFromJavaToCpp() called");
+
+    JNIEnv * env = chip::JniReferences::GetInstance().GetEnvForCurrentThread();
+    VerifyOrReturnValue(env != nullptr, nullptr,
+                        ChipLogError(AppServer, "convertTargetAppInfoFromJavaToCpp() Could not get JNIEnv for current thread"));
+    jclass targetAppInfoClass = env->GetObjectClass(jTargetAppInfo);
+    VerifyOrReturnValue(targetAppInfoClass != nullptr, nullptr,
+                        ChipLogError(AppServer, "convertTargetAppInfoFromJavaToCpp() TargetAppInfo class not found!"));
+
+    jfieldID vendorIdField  = env->GetFieldID(targetAppInfoClass, "vendorId", "Ljava/lang/Integer;");
+    jfieldID productIdField = env->GetFieldID(targetAppInfoClass, "productId", "Ljava/lang/Integer;");
+    VerifyOrReturnValue(vendorIdField != nullptr, nullptr,
+                        ChipLogError(AppServer, "convertTargetAppInfoFromJavaToCpp() vendorIdField not found!"));
+    VerifyOrReturnValue(productIdField != nullptr, nullptr,
+                        ChipLogError(AppServer, "convertTargetAppInfoFromJavaToCpp() productIdField not found!"));
+
+    jobject jVendorIdObject  = env->GetObjectField(jTargetAppInfo, vendorIdField);
+    jobject jProductIdObject = env->GetObjectField(jTargetAppInfo, productIdField);
+    VerifyOrReturnValue(jVendorIdObject != nullptr, nullptr,
+                        ChipLogError(AppServer, "convertTargetAppInfoFromJavaToCpp() vendorIdObject is null!"));
+    VerifyOrReturnValue(jProductIdObject != nullptr, nullptr,
+                        ChipLogError(AppServer, "convertTargetAppInfoFromJavaToCpp() productIdObject is null!"));
+
+    jclass integerClass      = env->FindClass("java/lang/Integer");
+    jmethodID intValueMethod = env->GetMethodID(integerClass, "intValue", "()I");
+
+    jint vendorId  = env->CallIntMethod(jVendorIdObject, intValueMethod);
+    jint productId = env->CallIntMethod(jProductIdObject, intValueMethod);
```

Comment:
You can still have the Product and VendorId default to null instead of 0 in the Java class by changing this JNI logic: if the jProductIdObject, instead of returning early from line 391, initialize jint productId to 0.

Reply:
Updating.